addons/mass_mailing/models/mailing_list.py

```diff
@@ -223,6 +224,13 @@ def action_merge(self, src_lists, archive):
     def close_dialog(self):
         return {'type': 'ir.actions.act_window_close'}
 
+    # ------------------------------------------------------
+    # MAILING
+    # ------------------------------------------------------
+
+    def _mailing_get_default_domain(self, mailing):
+        return [('list_ids', 'in', mailing.contact_list_ids.ids)]
```

Comment:
Your test case here below makes me wonder if it shouldn't be:

```
if mailing.contact_list_ids.ids:
    return [('list_ids', 'in', mailing.contact_list_ids.ids)]
else:
    return []
```

Or even something more agressive like (1, '!=', 1) to avoid fetching the whole database?

Reply:
[] means everything. Here we would have [('list_ids', 'in', [])], which returns no value, which is correct. Previously to this change actually a mailing list on "no list" means "mail everything" ^^ .